conans/client/source.py

```diff
@@ -206,6 +209,7 @@ def _run_scm(conanfile, src_folder, local_sources_path, output, cache):
         output.info("Getting sources from url: '%s'" % scm_data.url)
         scm = SCM(scm_data, dest_dir, output)
         scm.checkout()
+
     if cache:
         # This is a bit weird. Why after a SCM should we remove files. Maybe check conan 2.0
-        _clean_source_folder(dest_dir)
+        _clean_source_folder(dest_dir)  # TODO: Why removing in the cache? There is no danger.
```

Comment:
This is even more strange (as the previous comment states), but what can go wrong? I can't imagine any scenario, and I really like having the same behavior in the local workflow and inside the cache...

Reply:
I don't know either.

Reply:
It clears artifacts like ``conan_source.tgz`` in case they exist. It is very important to do so, otherwise they can be obsolete when uploading the package and contain other sources rather than the ones just processed.

Reply:
Maybe those files should be generated outside this folder, or removed after uploading. I think it would be safer than removing them in a command that has nothing to do with the one that generated them.